src/test/groovy/com/palantir/gradle/gitversion/GitVersionPluginTests.groovy

```diff
@@ -81,7 +81,39 @@ class GitVersionPluginTests extends Specification {
         git.runGitCommand("add", ".")
         git.runGitCommand("commit","-m", "'initial commit'")
         git.runGitCommand("tag", "-a", "1.0.0", "-m", "1.0.0")
-        println(projectDir)
+
+        when:
+        // will build the project at projectDir
+        BuildResult buildResult = with('printVersion').build()
+
+        then:
+        buildResult.output.contains(":printVersion\n1.0.0\n")
+    }
+
+    def 'git describe works when using worktree' () {
+        given:
+        File rootFolder = temporaryFolder
+        projectDir = Files.createDirectories(rootFolder.toPath().resolve('level1/newbranch')).toFile()
+        File initialDir = Files.createDirectories(rootFolder.toPath().resolve('level1/level2')).toFile()
```

Comment:
nit: Clean up this levels stuff, that's not what we're testing here

```suggestion
        projectDir = Files.createDirectories(rootFolder.toPath().resolve('original')).toFile()
        File initialDir = Files.createDirectories(rootFolder.toPath().resolve('worktree')).toFile()
```

Reply:
fixed based on the suggestion.